### ISSUE TITLE
Expand on "auxiliary" data like current request or user

### ISSUE THREAD

**Issue by MoritzS** [mjtamlyn/django-adapters#28]
I think we didn't explicitly mention how extra, non-ORM data should be handled.

On the construction step it says in the docs:
> It should be easy to modify the set of fields on a per-request basis.

In "Existing Data" it also says:
> This includes but is not limited to field defaults, initial data and model instances.

I still think we should explicitly document especially how the current request (and user) can be incorporated into the adapter since in my experience (reading #django) that is one the first things where new users get frustrated.

**Comment by mjtamlyn** [comment]
Agreed. In my opinion we should try to be explicit about context as well, rather than encouraging passing `request` - this makes testing more complex.
